setup.py

```diff
@@ -108,7 +108,15 @@ def finalize_options(self):
         build_ext.finalize_options(self)
         # Prevent numpy from thinking it is still in its setup process:
         # https://docs.python.org/2/library/__builtin__.html#module-__builtin__
-        __builtins__.__NUMPY_SETUP__ = False
+        try:
+            __builtins__.__NUMPY_SETUP__ = False
+        except:
+            try:
+                # For python 3
```

Comment:
Gensim does not support python 2 any more.

Reply:
This is a PR against 3.8.3 which does support Python 2. I am happy to remove Python 2 support from the PR if that is an issue. The problem this solves is, install with Poetry (Python 3) fails for the the same reason as reference in   https://github.com/RaRe-Technologies/gensim/pull/3363

Reply:
Yes, correct of course.

Reply:
I know it's not supported. But the Poetry install suddenly stopped working - same as in the develop branch. And for those of us that are still using it. This small PR fixes the issue.

Reply:
I think the best we can do is fix this on develop so future versions won't be affected.

Releasing a bugfix for 3.8.3 requires effort, the volume of which isn't justified, IMHO.